### PR TITLE
Add MVN-6 support to ngraph

### PR DESCRIPTION
**Pull request by mvafin**
Ticket: #30071

### CODE REVIEW

File: inference-engine/tests/functional/inference_engine/ngraph_reader/mvn_tests.cpp [hunk 1]
```diff
@@ -68,15 +80,28 @@ TEST_F(NGraphReaderTests, ReadMVNNetwork) {
                 </port>
             </output>
         </layer>
+		<layer name="Const_1" type="Const" precision="FP32" id="2">
```
Comment (ilyachur) [inline inference-engine/tests/functional/inference_engine/ngraph_reader/mvn_tests.cpp:83]: Several questions about tests:

- Could we keep tests for MVN from previous opsets?
- Why do you change the legacy IR? In legacy IR we didn't have `Const` layers and MVN layer didn't have `eps_mode` atribute.

File: ngraph/core/include/ngraph/op/mvn.hpp [hunk 2]
```diff
@@ -87,7 +88,55 @@ namespace ngraph
...
         using v0::MVN;
-    } // namespace op
-} // namespace ngraph
 
-NGRAPH_SUPPRESS_DEPRECATED_END
+        NGRAPH_SUPPRESS_DEPRECATED_END
+
+        namespace v6
+        {
+            enum class EpsMode
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/mvn.hpp:96]: Why do we need this enum?

File: ngraph/core/include/ngraph/op/util/attr_types.hpp [hunk 3]
```diff
@@ -259,6 +259,35 @@ namespace ngraph
...
+            // Apply eps outside sqrt
+            OUTSIDE_SQRT
+        };
+
+        NGRAPH_API
+        std::ostream& operator<<(std::ostream& s, const MVNEpsMode& type);
+    } // namespace op
+
+    template <>
+    class NGRAPH_API AttributeAdapter<op::MVNEpsMode>
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/util/attr_types.hpp:278]: I propose to have this `AttributeAdapter` in the same file with MVN operation.

File: ngraph/core/src/op/mvn.cpp [hunk 4]
```diff
@@ -117,3 +119,63 @@ bool op::MVN::visit_attributes(AttributeVisitor& visitor)
...
+    , m_eps{eps}
+    , m_normalize_variance{normalize_variance}
+    , m_eps_mode{eps_mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v6::MVN::validate_and_infer_types()
+{
+    set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
```
Comment (rkazants) [inline ngraph/core/src/op/mvn.cpp:162]: I missed some checks here: 1. `data` rank >= length(axes); 2. axes is 1d tensor. In more cases axes is a constant so we can check their values as well

**Review by rkazants — CHANGES_REQUESTED** [review]

File: ngraph/core/src/op/mvn.cpp [hunk 5]
```diff
@@ -28,6 +28,8 @@
 using namespace std;
 using namespace ngraph;
 
+// ------------------------------ V0 ------------------------------
+
 NGRAPH_SUPPRESS_DEPRECATED_START
```
[the remark this answers is not in the document]
Comment (mvafin) [reply ngraph/core/src/op/mvn.cpp:33]: It is there already

File: ngraph/core/include/ngraph/op/mvn.hpp [hunk 6]
```diff
@@ -87,7 +87,75 @@ namespace ngraph
...
+                MVN(const Output<Node>& data,
+                    const Output<Node>& reduction_axes,
+                    bool normalize_variance,
+                    float eps,
+                    MVNEpsMode eps_mode);
+
+                bool visit_attributes(AttributeVisitor& visitor) override;
+                void validate_and_infer_types() override;
+
+                virtual std::shared_ptr<Node>
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/mvn.hpp:134]: `virtual` is redundant

**Review by ilyachur — APPROVED** [review]

**Review by iimironov — APPROVED** [review]

**Comment by iefode** [comment]
Are you going to update ngraph reference implementation?

**Comment by mvafin** [comment]
> Are you going to update ngraph reference implementation?

Yes, but in different PR.

**Review by lazarevevgeny — APPROVED** [review]

**Review by rkazants — APPROVED** [review]

**Pull request merged**